R/preprocess.R

```diff
@@ -233,51 +345,173 @@ enw_reporting_triangle_to_long <- function(obs) {
   return(reports_long[])
 }
 
-#' @title FUNCTION_TITLE
-#' @description FUNCTION_DESCRIPTION
-#' @param obs PARAM_DESCRIPTION
+#' Construct preprocessed data
 #'
-#' @param by PARAM_DESCRIPTION, Default: c()
+#' This function is used internally by [enw_preprocess_data()] to combine
+#' various pieces of processed observed data into a single object. It
+#' is exposed to the user in order to allow for modular data preprocessing
+#' though this is not currently recommended. See documentation and code
+#' of [enw_preprocess_data()] for more on the expected inputs.
 #'
-#' @param max_delay PARAM_DESCRIPTION, Default: 20
+#' @param obs Observations with the addition of empirical reporting proportions
+#'  and and restricted to the specified maximum delay).
 #'
-#' @param ref_holidays DESCRIPTION
+#' @param new_confirm Incidence of notifications by reference and report date.
+#' Empirical reporting distributions are also added.
 #'
-#' @param rep_holidays DESCRIPTION
+#' @param latest The latest available observations.
 #'
-#' @param min_report_date PARAM_DESCRIPTION
+#' @param reporting_triangle Incident observations by report and reference
+#'  date in the standard reporting triangle matrix format.
 #'
-#' @param set_negatives_to_zero PARAM_DESCRIPTION, Default: TRUE
+#' @param metareference Metadata reference dates derived from observations.
 #'
-#' @return OUTPUT_DESCRIPTION
+#' @param metareport Metadata for report dates.
+#
+#' @inheritParams enw_preprocess_data
+#' @inherit enw_preprocess_data return
+#' @export
+#' @examples
+#' pobs <- enw_example("preprocessed")
+#' enw_construct_data(
+#'   obs = pobs$obs[[1]],
+#'   new_confirm = pobs$new_confirm[[1]],
+#'   latest = pobs$latest[[1]],
+#'   reporting_triangle = pobs$reporting_triangle[[1]],
+#'   metareport = pobs$metareport[[1]],
+#'   metareference = pobs$metareference[[1]],
+#'   max_delay = pobs$max_delay[[1]]
+#' )
+enw_construct_data <- function(obs, new_confirm, latest, reporting_triangle,
+                               metareport, metareference, max_delay) {
+  out <- data.table::data.table(
+    obs = list(obs),
+    new_confirm = list(new_confirm),
+    latest = list(latest),
+    reporting_triangle = list(reporting_triangle),
+    metareference = list(metareference),
+    metareport = list(metareport),
+    time = nrow(latest[group == 1]),
```

Comment:
I think this only works if all groups have the same reporting dates in the `obs` provided. But this is addressed in #64.

Reply:
yes, it does and yup. It is a bit of a shame we have this requirement but at least for now it makes things easier.